src/main/java/duke/task/DukeList.java

```diff
@@ -0,0 +1,234 @@
+import java.util.ArrayList;
+
+import duke.DukeException;
+
+
+/**
+ * Encapsulation of the list in Duke.
+ */
+public class DukeList {
+    /** The list in which Duke stores Tasks. */
+
+    private final ArrayList<Task> tasks = new ArrayList<>();
+
+
+    /**
+     * Constructs a DukeList.
+     */
+    public DukeList() {}
+
+    /**
+     * Displays the addition of a task.
+     *
+     * @param task The task to be displayed.
+     * @return The message associated to the task being displayed.
+     */
+    private String displayTask(Task task) {
+        String response = "Got it. I've added this task:\n";
+        String taskCount = "\nNow you have " + tasks.size() + " tasks in the list.";
+        return response + task.toString() + taskCount;
+    }
+
+    /**
+     * Adds a task to the list.
+     *
+     * @param text Description of the task to be added.
+     */
+    public void add(String text) {
+
+        tasks.add(new Task(text));
+        System.out.println("added: " + text);
+    }
+
+    /**
+     * Loads task data obtained from the save file into the list.
+     *
+     * @param type Type of task to be loaded.
+     * @param state Done state of the task to be loaded.
+     * @param body Description of the task to be loaded.
+     * @throws DukeException When unknown task type is loaded.
+     */
+    public void loadData(String type, String state, String body) throws DukeException {
+        Task task;
+        String[] sections;
+
+        switch (type) {
+        case "todo":
+            task = new ToDos(body);
+            break;
+        case "deadline":
+            sections = body.split(" /by ", 2);
+            task = new Deadlines(sections[0], sections[1]);
+            break;
+        case "event":
+            sections = body.split(" /at ", 2);
+            task = new Events(sections[0], sections[1]);
+            break;
+        default:
+            throw new DukeException("☹ OOPS!!! Unknown task type in saved data");
+        }
+
+        if (state.equals("1")) {
+            task.done();
+        }
+
+        this.tasks.add(task);
+    }
+
+    /**
+     * Adds a ToDos task to the list.
+     *
+     * @param text Body of the duke.task to be added.
+     * @return The message associated to the task being displayed.
+     * @throws DukeException If there is no text.
+     */
+    public String addToDo(String text) throws DukeException {
+        String message = text.trim();
+
+        if (message.equals("")) {
+            throw new DukeException("☹ OOPS!!! The description of a todo cannot be empty.");
+        }
+
+        ToDos input = new ToDos(message);
+
+        tasks.add(input);
+
+        return displayTask(input);
+    }
+
+    /**
+     * Adds a Deadlines task to the list.
+     *
+     * @param text Body of the task to be added.
+     * @return The message associated to the task being displayed.
+     */
+    public String addDeadlines(String text) throws DukeException {
```

Comment:
Probably `addDeadline ` is better?
Since plural form should be used on names representing a collection of objects.